src/test/utils/telemetry/TelemetryAggregator.test.ts

```diff
@@ -3,12 +3,201 @@
  *  Licensed under the MIT License. See LICENSE in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
+import { expect } from 'chai';
+import { TelemetryEventNotification } from 'vscode-languageserver-protocol';
+import { initEvent, TelemetryEvent } from '../../../client/TelemetryEvent';
+import { TelemetryAggregator } from '../../../service/utils/telemetry/TelemetryAggregator';
+import { TestConnection } from '../../TestConnection';
+
+const TelemetryAggregationInterval = 10;
+
 describe('(Unit) TelemetryAggregator', () => {
+    let testConnection: TestConnection;
+    let telemetryAggregator: TelemetryAggregator;
+    beforeEach('Prepare a language server for testing', async () => {
+        testConnection = new TestConnection();
+        telemetryAggregator = new TelemetryAggregator(testConnection.server, TelemetryAggregationInterval);
+    });
+
     describe('Common scenarios', () => {
-        xit('Should have tests');
+        it('Should return aggregated events', async () => {
+            const inputEvent1: TelemetryEvent = initEvent('aggregated');
+            inputEvent1.properties.canary = 'canary';
+            inputEvent1.measurements.duration = 5;
+
+            const inputEvent2: TelemetryEvent = initEvent('aggregated');
+            inputEvent2.properties.canary = 'canary';
+            inputEvent2.measurements.duration = 10;
+
+            const expected: TelemetryEvent = initEvent('aggregated');
+            expected.properties.canary = 'canary';
+            expected.measurements.count = 2;
+            expected.measurements.durationMedian = 7;
+            expected.measurements.durationMu = 1.956;
+            expected.measurements.durationSigma = 0.49;

```

Comment:
I miss C# [object initializers](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/how-to-initialize-objects-by-using-an-object-initializer)...